tests/reftests/pin.test

```diff
@@ -1,401 +1,911 @@
+### sh add-pin.sh nip-git git
+### sh add-pin.sh nip-path
+### sh add-pin.sh nip-path2
+### :::::::::::::::::::::::
+### :VI: Listing
+### :::::::::::::::::::::::
+### opam switch kinds
+### : pin list
+### opam pin list
+nip.1                         version           1
+nip-git.ved                   rsync             file://${BASEDIR}/nip-git
+nip-git2.ved                  git               git+file://${BASEDIR}/nip-git2#master
+nip-git3.ved                  git               git+file://${BASEDIR}/nip-git3#master
+nip-path.ved                  rsync             file://${BASEDIR}/nip-path
+nip-path3.ved                 rsync             file://${BASEDIR}/nip-path3
+no-url.1       (uninstalled)  local definition
+no-url2.1      (uninstalled)  local definition
+### : pin list --short
+### opam pin list --short
+nip
+nip-git
+nip-git2
+nip-git3
+nip-path
+nip-path3
+no-url
+no-url2
+### :::::::::::::::::::::::
+### :V: Scanning
+### :::::::::::::::::::::::
+### : pin scan
+### opam switch create scans --empty
+### opam pin scan ./nip-git | grep -v '^#'
+nip-git  ved        git+file://${BASEDIR}/nip-git#master  -
+### opam pin scan ./nip-path | grep -v '^#'
+nip-path  ved        file://${BASEDIR}/nip-path  -
+### mkdir lot_of_pkgs
+### sh -c "cp ./nip-*2/* lot_of_pkgs/"
+### sh -c "cp ./nip-*3/* lot_of_pkgs/"
+### opam pin scan ./lot_of_pkgs | grep -v '^#'
+nip-path3  ved        file://${BASEDIR}/lot_of_pkgs  -
+nip-path2  ved        file://${BASEDIR}/lot_of_pkgs  -
+nip-git3   ved        file://${BASEDIR}/lot_of_pkgs  -
+nip-git2   ved        file://${BASEDIR}/lot_of_pkgs  -
+### opam pin scan ./lot_of_pkgs --normalise
+nip-path3.ved^file://${BASEDIR}/lot_of_pkgs
+nip-path2.ved^file://${BASEDIR}/lot_of_pkgs
+nip-git3.ved^file://${BASEDIR}/lot_of_pkgs
+nip-git2.ved^file://${BASEDIR}/lot_of_pkgs
+### opam pin add nip-git2.ved^file://${BASEDIR}/lot_of_pkgs nip-git3.ved^file://${BASEDIR}/lot_of_pkgs | unordered
+This will pin the following packages: nip-git2, nip-git3. Continue? [y/n] y
+Package nip-git2 does not exist, create as a NEW package? [y/n] y
+[nip-git2.ved] synchronised (no changes)
+nip-git2 is now pinned to file://${BASEDIR}/lot_of_pkgs (version ved)
+Package nip-git3 does not exist, create as a NEW package? [y/n] y
+[nip-git3.ved] synchronised (no changes)
+nip-git3 is now pinned to file://${BASEDIR}/lot_of_pkgs (version ved)
 
 The following actions will be performed:
 === install 2 packages
-  - install bar dev (pinned)
-  - install qux dev (pinned)
+  - install nip-git2 ved (pinned)
+  - install nip-git3 ved (pinned)
 
-The following system packages will first need to be installed:
-    another-inexistant inexistant
+<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
+-> installed nip-git2.ved
+-> installed nip-git3.ved
+Done.
+### # shell on Cygwin ads a trialing `\r` in the middle of the string
+### sh -c 'opam pin add $(opam pin scan ./lot_of_pkgs --normalise | grep path | sed "s/\\r//g") ' | unordered
```

Comment:
What about:
```suggestion
### sh -c 'opam pin scan ./lot_of_pkgs --normalise | xargs opam pin add'
```

Reply:
it doesn't work neither. If there is 2 packages, the `\\r` issue is present

Reply:
interesting